src/utils/components/SelectTypeahead/SelectTypeahead.tsx

```diff
@@ -43,35 +44,40 @@ const SelectTypeahead: FC<SelectTypeaheadProps> = ({
   setInitialOptions,
   setSelected,
 }) => {
+  const { t } = useKubevirtTranslation();
   const [isOpen, setIsOpen] = useState(false);
   const [inputValue, setInputValue] = useState<string>(selected);
   const [filterValue, setFilterValue] = useState<string>('');
-  const [selectOptions, setSelectOptions] = useState<SelectOptionProps[]>(initialOptions);
+  const [selectOptions, setSelectOptions] = useState<SelectOptionProps[]>();
   const [focusedItemIndex, setFocusedItemIndex] = useState<null | number>(null);
   const [activeItemId, setActiveItemId] = useState<null | string>(null);
   const textInputRef = useRef<HTMLInputElement>();
 
   useEffect(() => {
-    if (isEmpty(initialOptions)) {
-      setSelectOptions([getCreateOption(filterValue, canCreate)]);
-      return;
+    const newSelectOptions: SelectOptionProps[] = filterValue
+      ? (initialOptions || [])?.filter((menuItem) =>
+          String(menuItem.children).toLowerCase().includes(filterValue.toLowerCase()),
+        )
+      : initialOptions || [];
+
+    if (canCreate) {
+      const createOption = getCreateOption?.(filterValue, canCreate);
+      if (createOption && !newSelectOptions.some((option) => option.value === CREATE_NEW)) {
+        newSelectOptions.push(createOption);
+      }
     }
-    let newSelectOptions: SelectOptionProps[] = initialOptions || [];
 
-    // Filter menu items based on the text input value when one exists
-    if (filterValue) {
-      newSelectOptions = initialOptions?.filter((menuItem) =>
-        String(menuItem.children).toLowerCase().includes(filterValue.toLowerCase()),
-      );
-
-      // If no option matches the filter exactly, display creation option
-      if (!initialOptions?.some((option) => option.value === filterValue) && canCreate) {
-        newSelectOptions = [...newSelectOptions, getCreateOption(filterValue, canCreate)];
-      }
+    if (
+      !canCreate &&
+      newSelectOptions.length === 0 &&
+      filterValue &&
+      !newSelectOptions.some((option) => option.value === NOT_FOUND)
```

Comment:
IIUC this option wouldn't never be reached if the second condition `newSelectOptions.length === 0` is true. Am I missing something?

Reply:
you're right, fixed